scene/main/node.cpp

```diff
@@ -217,8 +217,10 @@ void Node::_notification(int p_notification) {
 void Node::_propagate_ready() {
 	data.ready_notified = true;
 	data.blocked++;
-	for (KeyValue<StringName, Node *> &K : data.children) {
-		K.value->_propagate_ready();
+
+	_update_children_cache();
```

Comment:
While the fix is valid, this has severe performance consequences so it should not be made.
Instead, I suggest just checking for `data.blocked > 0`  during set_name() and suggest the user to use call_deferred("set_name"). instead.

Reply:
As a rule, calling _update_children_cache() can be very expensive, so it should only be done when accessing children by indices.

Reply:
Thanks for the insight. I have updated the PR according to your suggestions.